hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Desugarer.scala

```diff
@@ -89,11 +89,13 @@ class Desugarer(val elaborator: Elaborator)
     def ++(fallback: Split): Split =
       if fallback == Split.End then
         split
-      else (split match
+      else if split.isFull then
+        split
+      else split match
         case Split.Cons(head, tail) => Split.Cons(head, tail ++ fallback)
         case Split.Let(name, term, tail) => Split.Let(name, term, tail ++ fallback)
-        case Split.Else(_) => split
-        case Split.End => fallback)
+        case Split.Else(_) => die
```

Comment:
Ok, now I see that this was supposed to be impossible because of the `split.isFull` check that you added back...

It's not very good style anyway. I'd prefer just:
```suggestion
        case Split.Else(_) => split // Shouldn't actually happen because of the `split.isFull` check above
```
This way, if we refactor the code and this case actually does trigger, it will still work and not crash.